actionpack/lib/action_controller/metal/redirecting.rb

```diff
@@ -56,17 +60,14 @@ module Redirecting
     # To terminate the execution of the function immediately after the +redirect_to+, use return.
     #   redirect_to post_url(@post) and return
     def redirect_to(options = {}, response_options = {})
-      raise ActionControllerError.new("Cannot redirect to nil!") unless options
-      raise AbstractController::DoubleRenderError if response_body
+      response_options[:allow_other_host] ||= _allow_other_host unless response_options.key?(:allow_other_host)
```

Comment:
```suggestion
      response_options[:allow_other_host] ||= false if other_hosts_forbidden
```

Reply:
`response_options[:allow_other_host] ||= false` would be a no-op if it is unset, `nil` and `false` are both falsy values.